File: javascript/leetcode/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    var consec = 0
    var maxConsec = 0
    for (var i = 0; i < nums.length; i++) {
        if (nums[i] == 1) {
            consec++
        } else {
            if (consec > maxConsec) { maxConsec = consec}
            consec = 0
        }
    }
    if (consec > maxConsec) { maxConsec = consec}
    return maxConsec
};

File: ruby/leetcode/max-consecutive-ones.rb
def find_max_consecutive_ones(nums)
    consec = 0
    max_consec = 0
    for i in 0...nums.size()
        if nums[i] == 1
            consec += 1
        else
            max_consec = consec if consec > max_consec
            consec = 0
        end
    end
    max_consec = consec if consec > max_consec
    return max_consec
end
